### ISSUE TITLE
Enforce required with an association

### ISSUE THREAD

**Issue by Siecje** [balderdashy/sails#2822]
If an association is required then you shouldn't be able to specify a foreign key that doesn't exist to create the model.

The foreign key being used should exist.

Currently the foreign key is used in the database row and **when it does exist the association is created**.

I don't believe this is the intended behaviour.

**Comment by tjwebb** [comment]
> should be able to specify a foreign key that doesn't exist to create the model.

Can you elaborate on this? I'm not sure what you mean

**Comment by Siecje** [comment]
Typo ```shouldn't``` not ```should```.

If I want to users to have a company association I can set that the field is required but I can use a company id that doesn't exist and the user will be created.

If I use the company id of 1, when the first company is created the user will be associated with that company.

**Comment by tjwebb** [comment]
> but if the request contains a company id that doesn't exist the user will still be created.

Waterline should only automatically create new objects if you supply the entire object. Please share the actual objects you are creating, and the results

**Comment by Siecje** [comment]
The company field is supplied but the value (foreign key) it has does not exist.
I created a minimal project to show what is happening.

https://github.com/Siecje/sails-bug

The readme contains the steps that I took and what is happening.

**Comment by igorteton** [comment]
+1 
Just experienced this myself:
- Sails: 0.11.0
- sails-mongo: 0.10.6 (thought I don't think this matters as I can see other's are experiencing it with pgsql as well)

This seems really like a serious flaw...

Is there maybe some flag we need to set in config in order for models not be created if there is error returned back?

**Comment by Siecje** [comment]
I don't get an error, the new object is created and a valid response is returned (I assume 201).

**Comment by CWyrtzen** [comment]
Is this still an issue?

**Comment by sailsbot** [comment]
Thanks for posting, @Siecje.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
